charts/openshift-metering/templates/presto/presto-common-config.yaml

```diff
@@ -8,8 +8,14 @@ data:
     set -e
 
 {{- if .Values.presto.spec.presto.config.tls.enabled }}
-    openssl pkcs12 -export -inkey $PRESTO_KEYFILE -in $PRESTO_CRTFILE -out $PRESTO_KEYSTORE_PKCS12 -password pass:$PRESTO_KEYSTORE_PASSWORD
-    keytool -importkeystore -noprompt -srckeystore $PRESTO_KEYSTORE_PKCS12 -srcstoretype pkcs12 -destkeystore $PRESTO_KEYSTORE_JKS -storepass $PRESTO_KEYSTORE_PASSWORD -srcstorepass $PRESTO_KEYSTORE_PASSWORD
+    # copy all necessary tls.crt/tls.key from read-only secret volumes
+    cat $PRESTO_SERVER_KEYFILE $PRESTO_SERVER_CRTFILE > $PRESTO_SERVER_PEM
+    cat $PRESTO_SERVER_PEM $PRESTO_CA_CRTFILE > $PRESTO_KEYSTORE_PEM
+    cat $PRESTO_SERVER_PEM $PRESTO_CA_CRTFILE > $PRESTO_TRUSTSTORE_PEM
+{{- end }}
+{{- if .Values.presto.spec.presto.config.auth.enabled }}
+    # append the client key/certificate to the beginning of the truststore PEM
+    sed -i -e '1 e cat $PRESTO_CLIENT_KEYFILE $PRESTO_CLIENT_CRTFILE' $PRESTO_TRUSTSTORE_PEM
```

Comment:
I don't understand the sed expression here. Wan't we just concat things in the correct order?

Reply:
The sed expression above appends the values stored in `$PRESTO_CLIENT_*` to the beginning of the truststore key in order to maintain the certificate chain. I just used this for its brevity over the usage of `cat to re-construct the truststore, but I can change this to using the latter if you prefer the readability.